nightly build failure

```
 Compiling clippy_lints v0.0.198
error[E0599]: no method named `to_raw_bits` found for type `rustc::middle::const_val::ConstVal<'_>` in the current scope
   --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\consts.rs:212:48
    |
212 |                     ty::TyArray(_, n) => n.val.to_raw_bits().expect("array length"),
    |                                                ^^^^^^^^^^^

error[E0308]: mismatched types
   --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\consts.rs:420:25
    |
420 |         ConstVal::Value(Value::ByVal(PrimVal::Bytes(b))) => match result.ty.sty {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `rustc::mir::interpret::ConstValue`, found enum `rustc::mir::interpret::Value`
    |
    = note: expected type `rustc::mir::interpret::ConstValue<'_>`
               found type `rustc::mir::interpret::Value`

error[E0308]: mismatched types
   --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\consts.rs:428:25
    |
428 |         ConstVal::Value(Value::ByValPair(PrimVal::Ptr(ptr), PrimVal::Bytes(n))) => match result.ty.sty {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `rustc::mir::interpret::ConstValue`, found enum `rustc::mir::interpret::Value`
    |
    = note: expected type `rustc::mir::interpret::ConstValue<'_>`
               found type `rustc::mir::interpret::Value`

error[E0599]: no method named `to_raw_bits` found for type `rustc::middle::const_val::ConstVal<'_>` in the current scope
  --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\array_indexing.rs:64:37
   |
64 |                 let size = size.val.to_raw_bits().unwrap();
   |                                     ^^^^^^^^^^^

error[E0599]: no method named `to_raw_bits` found for type `rustc::middle::const_val::ConstVal<'_>` in the current scope
    --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\loops.rs:1226:72
     |
1226 |                         ty::TypeVariants::TyArray(_, size) if size.val.to_raw_bits().expect("array size") > 32 => (),
     |                                                                        ^^^^^^^^^^^

error[E0599]: no method named `to_raw_bits` found for type `rustc::middle::const_val::ConstVal<'_>` in the current scope
    --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\loops.rs:1801:55
     |
1801 |         ty::TyArray(_, n) => (0..=32).contains(&n.val.to_raw_bits().expect("array length")),
     |                                                       ^^^^^^^^^^^

error[E0599]: no method named `to_raw_bits` found for type `rustc::middle::const_val::ConstVal<'_>` in the current scope
    --> C:\Users\jaxx\.cargo\registry\src\github.com-1ecc6299db9ec823\clippy_lints-0.0.198\src\methods.rs:1302:46
     |
1302 |             ty::TyArray(_, size) => size.val.to_raw_bits().expect("array length") < 32,
     |                                              ^^^^^^^^^^^

error: aborting due to 7 previous errors

Some errors occurred: E0308, E0599.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,9 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## 0.0.199
+* Rustup to *rustc 1.27.0-nightly (ff2ac35db 2018-05-12)*
+
 ## 0.0.198
 * Rustup to *rustc 1.27.0-nightly (acd3871ba 2018-05-10)*
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "clippy"
-version = "0.0.198"
+version = "0.0.199"
 authors = [
 	"Manish Goregaokar <manishsmail@gmail.com>",
 	"Andre Bogus <bogusandre@gmail.com>",
@@ -37,7 +37,7 @@ path = "src/driver.rs"
 
 [dependencies]
 # begin automatic update
-clippy_lints = { version = "0.0.198", path = "clippy_lints" }
+clippy_lints = { version = "0.0.199", path = "clippy_lints" }
 # end automatic update
 regex = "1"
 semver = "0.9"

--- a/clippy_lints/Cargo.toml
+++ b/clippy_lints/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "clippy_lints"
 # begin automatic update
-version = "0.0.198"
+version = "0.0.199"
 # end automatic update
 authors = [
 	"Manish Goregaokar <manishsmail@gmail.com>",

--- a/clippy_lints/src/array_indexing.rs
+++ b/clippy_lints/src/array_indexing.rs
@@ -61,7 +61,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ArrayIndexing {
             // Array with known size can be checked statically
             let ty = cx.tables.expr_ty(array);
             if let ty::TyArray(_, size) = ty.sty {
-                let size = size.val.to_raw_bits().unwrap();
+                let size = size.assert_usize(cx.tcx).unwrap().into();
 
                 // Index is a constant uint
                 if let Some((Constant::Int(const_index), _)) = constant(cx, index) {

--- a/clippy_lints/src/consts.rs
+++ b/clippy_lints/src/consts.rs
@@ -209,7 +209,7 @@ impl<'c, 'cc> ConstEvalLateContext<'c, 'cc> {
             ExprTup(ref tup) => self.multi(tup).map(Constant::Tuple),
             ExprRepeat(ref value, _) => {
                 let n = match self.tables.expr_ty(e).sty {
-                    ty::TyArray(_, n) => n.val.to_raw_bits().expect("array length"),
+                    ty::TyArray(_, n) => n.assert_usize(self.tcx).expect("array length"),
                     _ => span_bug!(e.span, "typeck error"),
                 };
                 self.expr(value).map(|v| Constant::Repeat(Box::new(v), n as u64))
@@ -415,9 +415,9 @@ impl<'c, 'cc> ConstEvalLateContext<'c, 'cc> {
 }
 
 pub fn miri_to_const<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, result: &ty::Const<'tcx>) -> Option<Constant> {
-    use rustc::mir::interpret::{Value, PrimVal};
+    use rustc::mir::interpret::{PrimVal, ConstValue};
     match result.val {
-        ConstVal::Value(Value::ByVal(PrimVal::Bytes(b))) => match result.ty.sty {
+        ConstVal::Value(ConstValue::ByVal(PrimVal::Bytes(b))) => match result.ty.sty {
             ty::TyBool => Some(Constant::Bool(b == 1)),
             ty::TyUint(_) | ty::TyInt(_) => Some(Constant::Int(b)),
             ty::TyFloat(FloatTy::F32) => Some(Constant::F32(f32::from_bits(b as u32))),
@@ -425,7 +425,7 @@ pub fn miri_to_const<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, result: &ty::Const<'
             // FIXME: implement other conversion
             _ => None,
         },
-        ConstVal::Value(Value::ByValPair(PrimVal::Ptr(ptr), PrimVal::Bytes(n))) => match result.ty.sty {
+        ConstVal::Value(ConstValue::ByValPair(PrimVal::Ptr(ptr), PrimVal::Bytes(n))) => match result.ty.sty {
             ty::TyRef(_, tam, _) => match tam.sty {
                 ty::TyStr => {
                     let alloc = tcx

--- a/clippy_lints/src/loops.rs
+++ b/clippy_lints/src/loops.rs
@@ -1223,7 +1223,7 @@ fn check_for_loop_arg(cx: &LateContext, pat: &Pat, arg: &Expr, expr: &Expr) {
                     match cx.tables.expr_ty(&args[0]).sty {
                         // If the length is greater than 32 no traits are implemented for array and
                         // therefore we cannot use `&`.
-                        ty::TypeVariants::TyArray(_, size) if size.val.to_raw_bits().expect("array size") > 32 => (),
+                        ty::TypeVariants::TyArray(_, size) if size.assert_usize(cx.tcx).expect("array size") > 32 => (),
                         _ => lint_iter_method(cx, args, arg, method_name),
                     };
                 } else {
@@ -1784,7 +1784,7 @@ fn is_ref_iterable_type(cx: &LateContext, e: &Expr) -> bool {
     // no walk_ptrs_ty: calling iter() on a reference can make sense because it
     // will allow further borrows afterwards
     let ty = cx.tables.expr_ty(e);
-    is_iterable_array(ty) ||
+    is_iterable_array(ty, cx) ||
     match_type(cx, ty, &paths::VEC) ||
     match_type(cx, ty, &paths::LINKED_LIST) ||
     match_type(cx, ty, &paths::HASHMAP) ||
@@ -1795,10 +1795,10 @@ fn is_ref_iterable_type(cx: &LateContext, e: &Expr) -> bool {
     match_type(cx, ty, &paths::BTREESET)
 }
 
-fn is_iterable_array(ty: Ty) -> bool {
+fn is_iterable_array(ty: Ty, cx: &LateContext) -> bool {
     // IntoIterator is currently only implemented for array sizes <= 32 in rustc
     match ty.sty {
-        ty::TyArray(_, n) => (0..=32).contains(&n.val.to_raw_bits().expect("array length")),
+        ty::TyArray(_, n) => (0..=32).contains(&n.assert_usize(cx.tcx).expect("array length")),
         _ => false,
     }
 }

--- a/clippy_lints/src/methods.rs
+++ b/clippy_lints/src/methods.rs
@@ -1299,7 +1299,7 @@ fn derefs_to_slice(cx: &LateContext, expr: &hir::Expr, ty: Ty) -> Option<sugg::S
             ty::TySlice(_) => true,
             ty::TyAdt(def, _) if def.is_box() => may_slice(cx, ty.boxed_ty()),
             ty::TyAdt(..) => match_type(cx, ty, &paths::VEC),
-            ty::TyArray(_, size) => size.val.to_raw_bits().expect("array length") < 32,
+            ty::TyArray(_, size) => size.assert_usize(cx.tcx).expect("array length") < 32,
             ty::TyRef(_, inner, _) => may_slice(cx, inner),
             _ => false,
         }

--- a/min_version.txt
+++ b/min_version.txt
@@ -1,7 +1,7 @@
-rustc 1.27.0-nightly (acd3871ba 2018-05-10)
+rustc 1.27.0-nightly (ff2ac35db 2018-05-12)
 binary: rustc
-commit-hash: acd3871ba17316419c644e17547887787628ec2f
-commit-date: 2018-05-10
+commit-hash: ff2ac35db93a80b2de5daa4f280bf1503d62c164
+commit-date: 2018-05-12
 host: x86_64-unknown-linux-gnu
 release: 1.27.0-nightly
 LLVM version: 6.0

